managed-scripts/groovy/enableShallowCloneOnAllAbstractProjects.groovy

```diff
@@ -0,0 +1,22 @@
+import hudson.model.AbstractProject
+import hudson.plugins.git.GitSCM
+import hudson.plugins.git.extensions.impl.*
+
+Jenkins.instance.getAllItems(AbstractProject.class)
+.findAll { job -> job.isBuildable()}
+.findAll { job -> job.scm != null && job.scm instanceof GitSCM}
+.each { project -> 
+  scm = project.scm
+  
+  scm.extensions
+  .findAll {it instanceof CloneOption}
```

Comment:
So this actually modifies *existing `CloneOption` extension* already added to the GitSCM. What if there is no `CloneOption` extension ?

```
def cloneOption = scm.extensions.find {it instanceof CloneOption}
if (!cloneOption) {
    cloneOption = new CloneOption(true, false, "", 10)
} else {
    cloneOption = new CloneOption(true, it.noTags, it.reference, it.timeout)
}
scm.extensions.remove(it)
scm.extensions.add(cloneOption) 
```